src/main/java/com/buuz135/industrial/item/LaserLensItem.java

```diff
@@ -17,41 +19,63 @@
-    }
-
-    @Override
-    public void register(IForgeRegistry<Item> items) {
-        super.register(items);
-    }
-
-    public void createRecipe() {
-        for (int i = 0; i < 16; ++i)
-            RecipeUtils.addShapedRecipe(new ItemStack(this, 1, i), " i ", "ipi", " i ", 'i', inverted ? new ItemStack(Items.IRON_INGOT) : new ItemStack(ItemRegistry.pinkSlime),
-                    'p', new ItemStack(Blocks.STAINED_GLASS_PANE, 1, i));
-    }
-
-    @Override
-    public void registerRender() {
-        for (int i = 0; i < 16; ++i)
-            ModelLoader.setCustomModelResourceLocation(this, i, new ModelResourceLocation(this.getRegistryName().toString() + i, "inventory"));
-    }
-
-    @Override
-    public String getItemStackDisplayName(ItemStack stack) {
-        return new TextComponentTranslation("item.fireworksCharge." + EnumDyeColor.byMetadata(stack.getMetadata()).getUnlocalizedName().replaceAll("_", "")).getFormattedText() + " " + super.getItemStackDisplayName(stack);
-    }
+	@Getter
+	private boolean inverted;
+
+	public LaserLensItem(boolean inverted) {
+		super("laser_lens" + (inverted ? "_inverted" : ""));
+		setMaxStackSize(1);
+		setHasSubtypes(true);
+		this.inverted = inverted;
+	}
+
+	@Override
+	public void getSubItems(CreativeTabs tab, NonNullList<ItemStack> subItems) {
+		int nLens = CustomConfiguration.enableAdditionalLens ? 64 : 16;
+		if (isInCreativeTab(tab)) {
+			for (int i = 0; i < nLens; ++i) {
+				subItems.add(new ItemStack(this, 1, i));
+			}
+		}
+	}
+
+	@Override
+	public void register(IForgeRegistry<Item> items) {
+		super.register(items);
+	}
+
+	@Override
+	public void createRecipe() {
+		for (int i = 0; i < 16; ++i) {
+			ItemStack currentLens = new ItemStack(this, 1, i);
+			RecipeUtils.addShapedRecipe(currentLens, " i ", "ipi", " i ", 'i', inverted ? new ItemStack(Items.IRON_INGOT) : new ItemStack(ItemRegistry.pinkSlime),
+					'p', new ItemStack(Blocks.STAINED_GLASS_PANE, 1, i));
+			if (CustomConfiguration.enableAdditionalLens) {
+				RecipeUtils.addShapelessRecipe(new ItemStack(this, 1, i + 16), currentLens, Items.COAL);
+				RecipeUtils.addShapelessRecipe(new ItemStack(this, 1, i + 32), currentLens, Items.GOLD_INGOT);
+				RecipeUtils.addShapelessRecipe(new ItemStack(this, 1, i + 48), currentLens, Items.DIAMOND);
+			}
+		}
+	}
+
+	@Override
+	public void registerRender() {
+		int nLens = CustomConfiguration.enableAdditionalLens ? 64 : 16;
+		for (int i = 0; i < nLens; ++i) {
+			ModelLoader.setCustomModelResourceLocation(this, i, new ModelResourceLocation(this.getRegistryName().toString() + (i % 16), "inventory"));
+		}
+	}
+
+	@Override
+	public String getItemStackDisplayName(ItemStack stack) {
+		String color = new TextComponentTranslation("item.fireworksCharge." + EnumDyeColor.byMetadata(stack.getMetadata() % 16).getUnlocalizedName().replaceAll("_", "")).getFormattedText() + " ";
+		String coating = "";
+		if (stack.getMetadata() >= 48)
```

Comment:
This shouldn't be hardcoded, and it should use lang

Reply:
Ah, I see, I can fix that.